Classes/Bag/AbstractBag.php

```diff
@@ -164,14 +153,39 @@ private function executeQueryIfNotDoneYet()
             return;
         }
 
-        $this->queryResult = \Tx_Oelib_Db::selectMultiple(
-            static::$tableName . '.*',
-            $this->allTableNames,
-            $this->queryParameters . $this->enabledFieldsQuery,
-            $this->groupBy,
-            $this->orderBy,
-            $this->limit
-        );
+        /** @var ConnectionPool $pool */
+        $pool = GeneralUtility::makeInstance(ConnectionPool::class);
+        $queryBuilder = $pool->getQueryBuilderForTable($this->allTableNames);
+        foreach (preg_split('/,\\s*/', $this->allTableNames) as $tableName) {
+            $queryBuilder->from($tableName);
+        }
+
+        if ($this->showHiddenRecords) {
+            $queryBuilder
+                ->getRestrictions()
+                ->removeByType(HiddenRestriction::class)
+                ->removeByType(StartTimeRestriction::class)
+                ->removeByType(EndTimeRestriction::class);
+        }
+
+        $queryBuilder
+           ->select(static::$tableName . '.*')
+           ->where($this->queryParameters);
+
+        if ($this->groupBy) {
+            $queryBuilder->groupBy($this->groupBy);
+        }
+
+        if ($this->orderBy) {
+            [$field, $order] = explode(' ', $this->orderBy);
```

Comment:
This construct is not valid in PHP 7.0 (only in PHP >= 7.1).

Reply:
I know. I used `list()` but somehow my IDE always changes this into `[]`.